_posts/2021-12-13-why-we-should-rewrite-enzymejs-to-rtl-and-how-to-do-that.md

```diff
@@ -0,0 +1,333 @@
+    // when
+    render(<Test />);
+
+    // then
+    expect(screen.getByRole('heading', { name: 'Default header' })).toBeInTheDocument();
+  });
+
+  it('should render header from props', () => {
+    // when
+    render(<Test header="My simple calculator" />);
+
+    // then
+    expect(screen.getByRole('heading', { name: 'My simple calculator' })).toBeInTheDocument();
+    expect(screen.queryByText('Default header')).not.toBeInTheDocument();
+  });
+
+  it('should render default defaultCounter', () => {
+    // when
+    render(<Test />);
+
+    // then
+    expect(screen.getByText('0')).toBeInTheDocument();
+  });
+
+  it('should render defaultCounter from props', () => {
+    // when
+    render(<Test defaultCounter={100} />);
+
+    // then
+    expect(screen.getByText('100')).toBeInTheDocument();
+  });
+
+  it('should increment counter on click', () => {
+    // given
+    render(<Test defaultCounter={100} />);
+
+    const button = screen.getByRole('button', { name: 'Increment counter' });
+
+    // when
+    userEvent.click(button);
+
+    // then
+    expect(screen.getByText('101')).toBeInTheDocument();
+  });
+
+  it('should decrement counter on click', () => {
+    // given
+    render(<Test defaultCounter={100} />);
+
+    const button = screen.getByRole('button', { name: 'Decrement counter' });
+
+    // when
+    userEvent.click(button);
+
+    // then
+    expect(screen.getByText('99')).toBeInTheDocument();
+  });
+});
+
+```
+
+## Enzyme and RTL in Allegro Ads
+
+My team is currently working on the development of [Allegro Ads](https://allegro.pl/ads), which is an advertising tool
+ for our e-commerce platform sellers. It is a whole panel for viewing statistics, displaying advertising campaigns. We
+  provide sponsored offers for our Allegro site, Facebook, Google and recently even the partner websites. Our sellers
+   pay us for the opportunity to advertise their products, and at the same time we strive to ensure that these ads
+    accurately reach people potentially interested in buying, thereby generating profits for sellers.
+
+The main technology stack is standard and still pretty cool. The core of the project is written in React, and
+ significant parts have been rewritten to Typescript. For testing purposes, we use Jest. In my opinion this is a very
+  good and stable tool, easy to work with. It has quite a low entry threshold, even if someone hasn’t had too much
+   experience with tests before. The main advantages of Jest for me are minimal configuration and quite good
+    documentation. But for testing react components we currently have two libraries and I’ll explain why in a moment.
+
+The 2.0 version of platform Allegro Ads was released about 6 years ago. A year ago
+ the team added a React Testing Library to `package.json`. We have both packages so far. **Recently we officially
+  decided to slowly rewrite the Enzyme library for good.** We added the rule to our Declaration of Code Convention
```

Comment:
```suggestion
  decided to slowly rewrite the Enzyme tests and get rid of the library for good.** We added the rule to our Declaration of Code Convention
```

Może coś takiego? 
Nie przepisujemy samego Enzyme'a, ale napisane w nim testy, chociaż to Enzyme'a chcemy się pozbyć 😉